File: Backend/vs/src/main/java/com/foodapp/backend/service/CartService.java
package com.foodapp.backend.service;

import com.foodapp.backend.dto.cart.CartItemRequest;
import com.foodapp.backend.dto.cart.CartResponse;
import com.foodapp.backend.dto.cart.UpdateCartItemRequest;

public interface CartService {
    CartResponse getMyCart();

    CartResponse addItem(CartItemRequest request);

    CartResponse updateItem(Long cartItemId, UpdateCartItemRequest request);

    CartResponse removeItem(Long cartItemId);

    CartResponse clearCart();
}

File: Frontend/food-ordering-system/src/services/cartService.js
import axiosInstance from './axiosInstance';

const cartService = {
  getCart: () => axiosInstance.get('/cart'),
  addItem: (data) => axiosInstance.post('/cart/items', data),
  updateItem: (cartItemId, data) => axiosInstance.put(`/cart/items/${cartItemId}`, data),
  removeItem: (cartItemId) => axiosInstance.delete(`/cart/items/${cartItemId}`),
  clearCart: () => axiosInstance.delete('/cart'),
};

export default cartService;
